Camb vertex

Saransh-Shankar opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `uv run make format; uv run make lint` to appease the lint gods

Saransh-Shankar committed:
Integrated vertex AI

Signed-off-by: Saransh-Shankar <saransh.shankar@camb.ai>

Saransh-Shankar committed:
Merge branch 'main' of https://github.com/run-llama/llama_index into camb-vertex

Saransh-Shankar committed:
Made Fixes

Signed-off-by: Saransh-Shankar <saransh.shankar@camb.ai>

Saransh-Shankar committed:
Removed try-catch

Signed-off-by: Saransh-Shankar <saransh.shankar@camb.ai>

Saransh-Shankar committed:
Merge branch 'main' of https://github.com/run-llama/llama_index into camb-vertex

Saransh-Shankar committed:
Remove uv.lock

Signed-off-by: Saransh-Shankar <saransh.shankar@camb.ai>

arnavmehta7 commented on llama-index-core/tests/memory/blocks/test_static.py:
```diff
@@ -10,7 +15,9 @@ def sample_messages():
     """Create sample chat messages."""
     return [
         ChatMessage(role=MessageRole.USER, content="Hello, how are you?"),
-        ChatMessage(role=MessageRole.ASSISTANT, content="I'm doing well, thanks for asking!"),
```
undo changes to this file, you shouldn't ever change Opensource repo's default formatting

arnavmehta7 commented on llama-index-core/llama_index/core/storage/docstore/simple_docstore.py:
```diff
@@ -10,7 +10,10 @@
 )
 from llama_index.core.storage.kvstore.simple_kvstore import SimpleKVStore
-from llama_index.core.storage.kvstore.types import MutableMappingKVStore, BaseInMemoryKVStore
+from llama_index.core.storage.kvstore.types import (
```
why?

arnavmehta7 reviewed (approved):
make sure you are conident on formatting check if they recommend a formatter/use it

ColeMurray committed:
fix: prevent command injection vulnerability in RAG CLI --clear flag (#19322)

* fix: prevent command injection in RAG CLI clear command

- Replace os.system() with shutil.rmtree() to prevent command injection
- Add proper error handling for directory removal
- Fixes security vulnerability where malicious input in LLAMA_INDEX_CACHE_DIR
  could execute arbitrary commands when using --clear flag

The vulnerable code at line 201 used os.system() with user-controlled input,
allowing shell metacharacters to be interpreted. The fix uses shutil.rmtree()
which safely removes directories without invoking a shell.

* chore: remove security comment from code

AstraBert committed:
Adding an integration for LanceDB MultiModal AI LakeHouse (#19232)

logan-markewich committed:
Add firecrawl integration source (#19203)

logan-markewich committed:
fix missing raw in agent workflow events (#19325)

logan-markewich committed:
fix ensuring structured output (#19326)

AstraBert committed:
Deprecate Multi Modal LLMs (#19115)

ledongthuc committed:
Update concurrent_execution.md with None return (#19315)

logan-markewich committed:
add default max iterations of 20 for agents (#19035)

logan-markewich committed:
v0.12.47 (#19329)

Saransh-Shankar committed:
Merge branch 'main' of https://github.com/run-llama/llama_index into camb-vertex
